txstratum/protocol.py

```diff
@@ -151,8 +160,18 @@ def handle_result(self, result: Any, msgid: JSONRPCId) -> None:
         message = self.messages_in_transit.pop(msgid)
 
         if message.method == "client.get_version":
-            self.log.info("Miner version: {}".format(result), miner_id=self.miner_id)
+            if self.miner_version is None:
+                # Logs only the first time, otherwise this log would appear a lot because
+                # we use this method to estimate the RTT
+                self.log.info(
+                    "Miner version: {}".format(result), miner_id=self.miner_id
+                )
             self.miner_version = result
+            now = txstratum.time.time()
+            current_rtt = now - message.created_at
+            alpha = self.RTT_CONFIDENCE_CONSTANT
+            self.rtt = (1 - alpha) * self.rtt + alpha * current_rtt
+            self.rtt_message_in_transit = False
```

Comment:
I think there is some chance that we could never get a reply from the miner, right?

In this case, we would never set this to False, and consequently we would stopped sending new messages to it.

Should we use a date for this message instead of a boolean, to avoid this possibility?

We would set it to None here, and set it to the current date when sending a new one, and when checking this variable, we would check if it's not None and the date in it is not too old.

Does this make sense?

Reply:
Yes, it makes sense. Added a timeout of 60s https://github.com/HathorNetwork/tx-mining-service/pull/114/commits/9f8f4be1493e5cbe0892b4c63dc9a50bf23e0602